backend/models/friend.js

```diff
@@ -0,0 +1,36 @@
+module.exports = (sequelize, DataTypes) => {
+    const Friend = sequelize.define('Friend', {
```

Comment:
Why do we need a new Friend object? Aren't friends just other users? We should just need to do `User.hasMany(models.User, { as: 'Friends', foreignKey: 'userId'});`